### PR TITLE
Refactor JNI code out of Core

### PR DESCRIPTION
**Pull request by bmeike**
Large, multi-repo project to refactor the Java specific JNI code out of Core

See: https://github.com/couchbase/couchbase-lite-core/pull/716
See: https://github.com/couchbaselabs/couchbase-lite-android-ee/pull/33

### CODE REVIEW

**Comment by bmeike** [comment]
This change breaks `liteserv-android`.  I will open a separate ticket to either fix it or delete it.

File: shared/LiteCoreJNI.exp [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#  LiteCoreJNI.exp
+#  Couchbase Lite Core
+#
+#  Created by Jens Alfke on 9/14/15.
+#  Copyright (c) 2015-2016 Couchbase. All rights reserved.
```
Comment (pasin) [inline shared/LiteCoreJNI.exp]: The license header is not correct.

Comment (pasin) [reply shared/LiteCoreJNI.exp]: I'm also wondering if we need this file or not.

Comment (bmeike) [reply shared/LiteCoreJNI.exp]: Good catch.  Totally out of scope for this change.  I will open a separate ticket for it, though.

**Comment by bmeike** [comment]
This commit does not in any way change the test coverage.  I'm ignoring the coveralls warning.

File: shared/LiteCoreJNI.exp (hunk 1 above)
Comment (bmeike) [reply shared/LiteCoreJNI.exp]: Opened issue: https://github.com/couchbase/couchbase-lite-android/issues/1877

**Pull request merged**